### PR TITLE
fix(credentials): add missing issue credential v1 proposal attributes

### PR DESCRIPTION
**Pull request by NB-MikeRichardson**

### CODE REVIEW

File: packages/core/src/modules/credentials/__tests__/V1CredentialService.propose.offer.test.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import type { AgentConfig } from '../../../agent/AgentConfig'
```
Comment (TimoGlastra) [inline packages/core/src/modules/credentials/__tests__/V1CredentialService.propose.offer.test.ts:1]: It's getting quite crazy with the dots. Should we create a `V1CredentialService.proposal.test.ts` or should we maybe just call it `V1CredentialService.proposal-offer.test.ts` Ideally we remove the dots all together, it's more meant as a file type, while here it's just part of the filename IMO

File: packages/core/src/modules/credentials/__tests__/V1CredentialService.propose.offer.test.ts [hunk 2]
```diff
@@ -126,6 +126,108 @@ describe('CredentialService', () => {
...
+        },
+        '~attach': [
+          {
+            '@id': expect.any(String),
+            'mime-type': 'application/json',
+            data: {
+              base64: expect.any(String),
+            },
+          },
+        ],
```
Comment (TimoGlastra) [inline packages/core/src/modules/credentials/__tests__/V1CredentialService.propose.offer.test.ts:226]: Where is the attachment for? I don't think the v1 message should have an attachment.

File: packages/core/src/modules/credentials/__tests__/V2CredentialService.offer.test.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import type { AgentConfig } from '../../../agent/AgentConfig'
 import type { ConnectionService } from '../../connections/services/ConnectionService'
 import type { CredentialStateChangedEvent } from '../CredentialEvents'
-import type { OfferCredentialOptions } from '../CredentialsModuleOptions'
+import type { OfferCredentialOptions, ProposeCredentialOptions } from '../CredentialsModuleOptions'
```
Comment (TimoGlastra) [inline packages/core/src/modules/credentials/__tests__/V2CredentialService.offer.test.ts:4]: unused

File: packages/core/src/modules/credentials/protocol/v1/V1CredentialService.ts [hunk 4]
```diff
@@ -125,10 +125,10 @@ export class V1CredentialService extends CredentialService {
     }
 
     const config: CredentialProposeOptions = {
```
Comment (TimoGlastra) [inline packages/core/src/modules/credentials/protocol/v1/V1CredentialService.ts:127]: why are we calling the format service here? We don't use the attachment, is it for the linked attachments? I'm not sure

File: packages/core/src/modules/credentials/__tests__/V1CredentialService.propose.offer.test.ts (hunk 2 above)
Comment (NB-MikeRichardson) [reply packages/core/src/modules/credentials/__tests__/V1CredentialService.propose.offer.test.ts:226]: I have removed the attachment and it passes.

File: packages/core/src/modules/credentials/protocol/v1/V1CredentialService.ts (hunk 4 above)
Comment (NB-MikeRichardson) [reply packages/core/src/modules/credentials/protocol/v1/V1CredentialService.ts:127]: Yes it looks like the createProposal method in the format service adds the linked attachments to the proposal but it also validates the proposal payload...I assume this is required as it relates to the interoperability issue we hit last week.

Comment (TimoGlastra) [reply packages/core/src/modules/credentials/protocol/v1/V1CredentialService.ts:127]: Ok, could we maybe add a comment explaining this? Future readers of the code can maybe be confused that we aren't using the result

File: packages/core/src/modules/credentials/__tests__/V1CredentialService.propose.offer.test.ts (hunk 1 above)
Comment (NB-MikeRichardson) [reply packages/core/src/modules/credentials/__tests__/V1CredentialService.propose.offer.test.ts]: Files renamed

File: packages/core/src/modules/credentials/__tests__/V1CredentialServiceProposeOffer.test.ts [hunk 5]
```diff
@@ -126,6 +126,99 @@ describe('CredentialService', () => {
...
+            state: CredentialState.ProposalSent,
+          }),
+        },
+      })
+    })
+
+    test('returns credential proposal message', async () => {
+      const { message: credentialProposal } = await credentialService.createProposal(proposeOptions)
+
+      // console.log(credentialProposal.toJSON())
```
Comment (TimoGlastra) [inline packages/core/src/modules/credentials/__tests__/V1CredentialServiceProposeOffer.test.ts:193]:
```suggestion
```

File: packages/core/src/modules/credentials/__tests__/V1CredentialServiceProposeOffer.test.ts [hunk 6]
```diff
@@ -126,6 +126,99 @@ describe('CredentialService', () => {
...
+      expect(credentialProposal.toJSON()).toMatchObject({
```
Comment (TimoGlastra) [inline packages/core/src/modules/credentials/__tests__/V1CredentialServiceProposeOffer.test.ts:194]: 👍

**Review by TimoGlastra — APPROVED** [review]
If you can fix the merge conflicts we can merge, thanks!

**Pull request merged**